assert/numeric_guidance.go

```diff
@@ -117,7 +129,16 @@ func newGapValue[T numConstraint](sz T, is_neg bool) any {
 	case float64:
 		return gapValue[float64]{gap_size: float64(sz), gap_is_negative: is_neg}
 	}
-	return nil
+
+	// Got this far, and still not sure?
+	numVal := reflect.ValueOf(sz)
+	if numVal.CanUint() {
+		return gapValue[uint64]{gap_size: numVal.Uint(), gap_is_negative: is_neg}
+	} else if numVal.CanFloat() {
+		return gapValue[float64]{gap_size: numVal.Float(), gap_is_negative: is_neg}
+	} else {
+		return nil
```

Comment:
Is this necessary? If `sz` is constraint by `numConstraint`, which is just `uint64 | float64`, shouldn't the type switch right above already cover all cases? The reflection-based approach should only be necessary when we are dealing with constraints that talks about underlying types (like `~uint64 | ~float64`)?

Reply:
Yes, i believe that performing the reflection will not ever be needed here.  I'll take this out.